build/builder/Dockerfile

```diff
@@ -202,7 +204,8 @@ RUN ln -s /go/src/github.com/cockroachdb/cockroach/build/builder/mkrelease.sh /u
 
 RUN curl -fsSL https://github.com/benesch/autouseradd/releases/download/1.2.0/autouseradd-1.2.0-amd64.tar.gz -o autouseradd.tar.gz \
   && echo 'f7b0cf67564044c31ffc5e84c961726098b88b0296a57c84421659d56434a365 autouseradd.tar.gz' | sha256sum -c - \
-  && tar xzf autouseradd.tar.gz --strip-components 1 \
+  && tar xzf autouseradd.tar.gz -C / --strip-components 1 \
```

Comment:
This looks unnecessary. At `master`:

```
roach@6b1826db1f19:/go/src/github.com/cockroachdb/cockroach$ which autouseradd
/usr/local/bin/autouseradd
roach@6b1826db1f19:/go/src/github.com/cockroachdb/cockroach$ ls -lah /usr/local/bin/autouseradd
-rwxr-xr-x 1 root root 8.8K Jul  7  2020 /usr/local/bin/autouseradd
```

Reply:
Strange, when I build this image, I found the `autouseradd` located under the home directory, if I remember correctly, it was root's home directory.

From the Dockerfile code, you also can see the current working directory is always the home directory, never changed.

Reply:
If I did not add this change, the entrypoint would say it cannot find the `autouseradd` binary.

Reply:
If not adding this change:
```
% docker run -it crdb-build  
docker: Error response from daemon: OCI runtime create failed: container_linux.go:380: starting container process caused: exec: "autouseradd": executable file not found in $PATH: unknown.
ERRO[0000] error waiting for container: context canceled 
```
After adding this change:
```
% docker run -it crdb-build   
autouseradd: running as root is not permitted
```